### PR TITLE
retrofit v3

### PR DESCRIPTION
**Pull request by nbhd**
https://github.com/nbhd/welcome-to-android/issues/17
- [x] @shaunkawano 
- [x] @eoinfogarty 
- [x] @bowyer-app

### CODE REVIEW

File: grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java [hunk 1]
```diff
@@ -74,25 +74,38 @@ public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
...
+        api
+                .getRepositories(
+                        "language:java",
+                        "Repositories",
+                        ++mPage + "",
+                        PER_PAGE
+                )
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new Observer<Repositories>() {
```
Comment (shoheikawano) [inline grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java:96]: When you subscribe, you MUST unsubscribe. (unless what you subscribe is synchronous work) 

See below:
- https://github.com/ReactiveX/RxJava
- http://qiita.com/yuya_presto/items/c8c3d77ac958c9c8f67b

Comment (nbhd) [inline grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java:96]: oh, I forgot it.

File: grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    public GithubDao(GithubApi githubApi) {
+        mGithubApi = githubApi;
+    }
+
+    public Observable<Repositories> getRepositories(String q, String type, int page, int per_page) {
+        return mGithubApi
+                .getRepositories(
+                        q,
+                        type,
+                        page + "",
```
Comment (shoheikawano) [inline grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java]: This works fine. 👍  You may also use `String.valueOf(someValue);` http://www.techscore.com/blog/2012/11/28/%E6%95%B0%E5%80%A4-%E2%87%94-%E6%96%87%E5%AD%97%E5%88%97%E5%A4%89%E6%8F%9B-java%E7%B7%A8/

I think this is more common than concatenating empty string. 

If you are manipulating `String` values in Android, you can also use `TextUtils` class as well which is a utility class that Android platform provides.

https://developer.android.com/reference/android/text/TextUtils.html

File: grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import rx.Observable;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.schedulers.Schedulers;
+
+public class GithubDao {
+
+    private GithubApi mGithubApi;
+
+    public GithubDao(GithubApi githubApi) {
+        mGithubApi = githubApi;
```
Comment (eoinfogarty) [inline grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java]: Any reason to pass in the constructor?

`mGithubApi = ApiClient.getClient().create(GithubApi.class);`

File: grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java (hunk 2 above)
Comment (eoinfogarty) [inline grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java]: `String.valueOf(someValue)` is also null safe

Comment (eoinfogarty) [inline grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java]:
``` java
public static String valueOf(Object obj) {
      return (obj == null) ? "null" : obj.toString();
}
```

http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/String.java#String.valueOf%28java.lang.Object%29

Comment (eoinfogarty) [inline grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java]: TextUtils has a helpful `isEmpty` method

File: grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java (hunk 3 above)
Comment (bowyer-app) [inline grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java]: i think so too.

Comment (nbhd) [inline grepos/app/src/main/java/net/boondockradio/grepos/dao/GithubDao.java]: adopt!

**Comment by bowyer-app** [comment]
LGTM

**Comment by eoinfogarty** [comment]
LGTM

**Comment by shoheikawano** [comment]
LGTM

**Pull request merged**